tools/ci/src/prepare.rs

```diff
@@ -87,4 +96,43 @@ impl<'a> PreparedCommand<'a> {
         self.env_vars.push((key, value));
         self
     }
+
+    /// A builder that controls whetever this command outputs json
+    pub fn with_json(mut self, emit_json: bool) -> Self {
+        self.emit_json = emit_json;
+        self
+    }
+
+    /// Runs this command
+    ///
+    /// If the command otputs json will return a [`JsonCommandOutput`]
+    pub fn run(
+        &mut self,
+        shell: &Shell,
+    ) -> Result<Option<JsonCommandOutput>, Option<JsonCommandOutput>> {
+        // If the CI test is to be executed in a subdirectory, we move there before running the command.
+        // This will automatically move back to the original directory once dropped.
+        let _subdir_hook = self.subdir.map(|path| shell.push_dir(path));
+
+        // For json outputting commands we want to read stdout even when things fail.
+        self.command.set_ignore_status(true);
```

Comment:
An alternative to this is to only ignore the status when `emit_json` is true, though this would involve refactoring all of your logic.

```suggestion
        self.command.set_ignore_status(self.emit_json);
```

Reply:
I'm going to not do this since the code is getting replace soon anyway.